.github/workflows/test-external-providers.yml

```diff
@@ -0,0 +1,89 @@
+name: Test External Providers
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  test-external-providers:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
+        with:
+          python-version: "3.10"
+
+      - name: Install Ollama
+        run: |
+          curl -fsSL https://ollama.com/install.sh | sh
+
+      - name: Pull Ollama image
+        run: |
+          ollama pull llama3.2:3b-instruct-fp16
+
+      - name: Start Ollama in background
+        run: |
+          nohup ollama run llama3.2:3b-instruct-fp16 > ollama.log 2>&1 &
+
+      - name: Set Up Environment and Install Dependencies
+        run: |
+          uv sync --extra dev --extra test
+          uv pip install -e .
+
+      - name: Clone Ollama provider
+        run: |
+          git clone https://github.com/leseb/llama-stack-provider-ollama.git
```

Comment:
I don't think we should depend on a repo external to the organization for CI. Would it make sense to keep everything related to the test run in-tree? I think the main point is that the directory is outside the expected location; which can be configured without an external repo.

Reply:
Ok I added this as a separate commit in case maintainers think otherwise https://github.com/meta-llama/llama-stack/pull/1672/commits/f0a3c2e01f6e527f1db7e341cb98679d35eacb1f